### ISSUE TITLE
Bug: internal server error: POST /v1/token/authentication with user which does not exist

### ISSUE THREAD

**Issue by unknowntpo** [unknowntpo/todos#4]
Description:

1. When I sent a POST request to `/v1/users/registration` with body:

```
{
  "email": "bob@example.com",
  "name": "bob",
  "password": "pa55word"
}
```
2. get the activation token from my Mailtrap service.

3. Sent a POST request to `/v1/tokens/authentication` with
a user that is not stored in the user database, like:

```
{
  "email": "bod@example.com", 
  "password": "pa55word"
}
```

I got the following error response with status code 500:

```
{
  "error": "the server encountered a problem and could not process your request"
}
```

And the error logging:

```
{
  "level": "error",
  "stack": [
    {
      "func": "New",
      "line": "172",
      "source": "errors.go"
    },
    {
      "func": "init",
      "line": "8",
      "source": "errors.go"
    },
    {
      "func": "doInit",
      "line": "6315",
      "source": "proc.go"
    },
    {
      "func": "doInit",
      "line": "6292",
      "source": "proc.go"
    },
    {
      "func": "doInit",
      "line": "6292",
      "source": "proc.go"
    },
    {
      "func": "main",
      "line": "208",
      "source": "proc.go"
    },
    {
      "func": "goexit",
      "line": "1371",
      "source": "asm_amd64.s"
    }
  ],
  "error": "userUsecase.Login: userRepo.GetByEmail: record not found: record not found",
  "time": "2021-12-02T11:43:20Z"
}
```

**Comment by unknowntpo** [comment]
1. Stack trace message doesn't show the real location where an error happened, it just shows where we use the `errors.New()` function to create the variable `domain.ErrRecordNotFound`

```
{
  "level": "error",
  "stack": [
    {
      "func": "New",      // Where the domain/errors.New are defind
      "line": "172",
      "source": "errors.go"
    },
    {
      "func": "init",
      "line": "8",
      "source": "errors.go"
    },
    {
      "func": "doInit",
      "line": "6315",
      "source": "proc.go"
    },
    {
      "func": "doInit",
      "line": "6292",
      "source": "proc.go"
    },
    {
      "func": "doInit",
      "line": "6292",
      "source": "proc.go"
    },
    {
      "func": "main",
      "line": "208",
      "source": "proc.go"
    },
    {
      "func": "goexit",
      "line": "1371",
      "source": "asm_amd64.s"
    }
  ],
  "error": "userUsecase.Login: userRepo.GetByEmail: record not found: record not found",
  "time": "2021-12-02T11:43:20Z"
}
```

2. The real problem is, at user repo, we return an `errors.Error` with `errors.KindRecordNotFound`
https://github.com/unknowntpo/todos/blob/73c943c2e7323628cbcad74fcb82357440787777/internal/user/repository/postgres/postgres_repo.go#L63-L80

but in `tokenAPI`, we check the type is whether `errors.KindInvalidCredentials` or not, that's the problem! 
https://github.com/unknowntpo/todos/blob/73c943c2e7323628cbcad74fcb82357440787777/internal/token/delivery/api/token_api.go#L61-L72

I think the solution is:

For problem 1:

Change the error variable from `errors`, which does not contain stack trace message.
And `errors.E()` will add a stack trace at where we call `errors.E()`.

https://github.com/unknowntpo/todos/blob/73c943c2e7323628cbcad74fcb82357440787777/internal/domain/errors/errors.go#L222-L231

For problem 2: at `userUsecase.Login`, when there's an error from `userRepo.GetByEmail`, if the error is `domain.ErrRecordNotFound` , we add a `errors.KindInvalidCredentials` to indicate that it is an invalid user login.

```diff
func (uu *userUsecase) Login(ctx context.Context, email, password string) (*domain.Token, error) {
	const op errors.Op = "userUsecase.Login"

	ctx, cancel := context.WithTimeout(ctx, uu.contextTimeout)
	defer cancel()

        user, err := uu.userRepo.GetByEmail(ctx, email)
	if err != nil {
-          return nil, errors.E(op, err)
+          return nil, errors.E(op,errors.KindInvalidCredentials, err)
	}
   
        ...
}
```

**Comment by unknowntpo** [comment]
Solved by #7
